consensus/bor/heimdallapp/checkpoint.go

```diff
@@ -0,0 +1,50 @@
+package heimdallapp
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/ethereum/go-ethereum/consensus/bor/heimdall/checkpoint"
+	"github.com/ethereum/go-ethereum/log"
+
+	abci "github.com/tendermint/tendermint/abci/types"
+)
+
+func (h *HeimdallAppClient) FetchCheckpointCount(ctx context.Context) (int64, error) {
+	log.Info("Fetching checkpoint count")
+
+	hCtx := h.hApp.NewContext(true, abci.Header{Height: h.hApp.LastBlockHeight()})
+
+	res := h.hApp.CheckpointKeeper.GetACKCount(hCtx)
+
+	log.Info("Fetched checkpoint count")
+
+	return int64(res), nil
+}
+
+func (h *HeimdallAppClient) FetchCheckpoint(ctx context.Context, number int64) (*checkpoint.Checkpoint, error) {
+	log.Info("Fetching checkpoint", "number", number)
+
+	hCtx := h.hApp.NewContext(true, abci.Header{Height: h.hApp.LastBlockHeight()})
+
+	res, err := h.hApp.CheckpointKeeper.GetCheckpointByNumber(hCtx, uint64(number))
+	if err != nil {
+		return nil, err
+	}
+
+	log.Info("Fetched checkpoint", "number", number)
+
+	resBytes, err := json.Marshal(res)
+	if err != nil {
+		return nil, err
+	}
+
+	var checkpoint checkpoint.Checkpoint
+
+	err = json.Unmarshal(resBytes, &checkpoint)
```

Comment:
why we marshall to unmarshall then?

Reply:
To convert the response from `types.Checkpoint` to `checkpoint.Checkpoint`
Basically they have similar data but one is from heimdall repo (from where we are fetching checkpoint) and other is bor repo's checkpoint object.

Reply:
We can't directly assign as the data and json tags are same, But their internall types are different.

https://github.com/maticnetwork/heimdall/blob/develop/types/header.go#L9

https://github.com/maticnetwork/bor/blob/develop/consensus/bor/heimdall/checkpoint/checkpoint.go#L10